packages/flutter/lib/src/material/divider_theme.dart

```diff
@@ -0,0 +1,167 @@
+  ///
+  /// This represents the amount of horizontal or vertical space the divider
+  /// takes up.
+  final double space;
+
+  /// The thickness that the divider is drawn as.
+  final double thickness;
+
+  /// The amount of empty space to leading edge of [Divider] or top edge of
+  /// [VerticalDivider].
+  final double indent;
+
+  /// The amount of empty space to trailing edge of [Divider] or bottom edge of
+  /// [VerticalDivider].
+  final double endIndent;
+
+  /// Creates a copy of this object with the given fields replaced with the
+  /// new values.
+  DividerThemeData copyWith({
+    Color color,
+    double space,
+    double thickness,
+    double indent,
+    double endIndent,
+  }) {
+    return DividerThemeData(
+      color: color ?? this.color,
+      space: space ?? this.space,
+      thickness: thickness ?? this.thickness,
+      indent: indent ?? this.indent,
+      endIndent: endIndent ?? this.endIndent,
+    );
+  }
+
+  /// Linearly interpolate between two Divider themes.
+  ///
+  /// The argument `t` must not be null.
+  ///
+  /// {@macro dart.ui.shadow.lerp}
+  static DividerThemeData lerp(DividerThemeData a, DividerThemeData b, double t) {
+    assert(t != null);
+    return DividerThemeData(
+      color: Color.lerp(a?.color, b?.color, t),
+      space: lerpDouble(a?.space, b?.space, t),
+      thickness: lerpDouble(a?.thickness, b?.thickness, t),
+      indent: lerpDouble(a?.indent, b?.indent, t),
+      endIndent: lerpDouble(a?.endIndent, b?.endIndent, t),
+    );
+  }
+
+  @override
+  int get hashCode {
+    return hashValues(
+      color,
+      space,
+      thickness,
+      indent,
+      endIndent,
+    );
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    final DividerThemeData typedOther = other;
+    return typedOther.color == color
+        && typedOther.space == space
+        && typedOther.thickness == thickness
+        && typedOther.indent == indent
+        && typedOther.endIndent == endIndent;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(ColorProperty('color', color, defaultValue: null));
+    properties.add(DoubleProperty('space', space, defaultValue: null));
+    properties.add(DoubleProperty('thickness', thickness, defaultValue: null));
+    properties.add(DoubleProperty('indent', indent, defaultValue: null));
+    properties.add(DoubleProperty('endIndent', endIndent, defaultValue: null));
+  }
+}
+
+/// An inherited widget that defines the configuration for
+/// [Divider]s, [VerticalDividers]s, etc. in this widget's subtree.
+///
+/// Values specified here are used for [Divider]s, [VerticalDividers]s, etc.
+/// properties that are not given an explicit non-null value.
+class DividerTheme extends InheritedWidget {
```

Comment:
Depending on which lands first, this, or https://github.com/flutter/flutter/pull/38583, you may want to change this to `InheritedTheme`.

Reply:
good to know